pytest_operator/plugin.py

```diff
@@ -281,12 +301,48 @@ async def log_model(self):
         )
         log.info(f"Juju error logs:\n\n{stdout}")
 
+    async def create_crash_dump(self) -> bool:
+        """Run the juju-crashdump if it's possible."""
+        cmd = [
+            "juju-crashdump",
+            "-s",
+            "-m",
+            self.model_full_name,
+            "-a",
+            "debug-layer",
+            "-a",
+            "config",
+        ]
+
+        base_path = _source_charm_dir(self.tmp_path)
```

Comment:
I think this might be prone to breaking in some environments (such as Jenkins) where non-standard home directories conflict with snap permissions, leading to the temp dir being location outside of the path-under-test. I'm not certain if this is still an issue, but it seems like [`self.request.config.invocation_params.dir`](https://docs.pytest.org/en/6.2.x/reference.html#pytest.config.Config.InvocationParams.dir) would be a safe and easy alternative.

Note that both approaches, though, have issues when run via nested tox calls like used in some Kubeflow repos (e.g., [istio-operators](https://github.com/canonical/istio-operators/blob/master/tox.ini#L18)). Maybe it would be best still to just go with the current directory, or even add an explicit `--crash-dump-output` option?

Reply:
I like the idea to add explicit option or using current directory. What do you think about adding 3th option, the environment variable? `request.config.option.crash_dump_output or os.getenv("CRASH_DUMP_OUTPUT") or ","`

There is related [PR](https://github.com/charmed-kubernetes/actions-operator/pull/31) for `actions-operator`, where I used function to looked for any `**/juju-crashdump-*.tar.xz` so the environment variable is not necessary, but it would be more explicit to look into specific directory.